src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -132,11 +132,12 @@ private AssessmentDto toDto(Assessment assessment, Participant participant, bool
 
         List<Submission> participantSubmissions = submissionService.findByParticipantId(participant.getParticipantId());
```

Comment:
Suggestion: use submissionService.findByParticipantIdAndAssessmentId to do the assessment filtering here

Reply:
Thanks! Fixed.